File: programmers/약수의 개수와 덧셈.py
def solution(left, right):
    answer = 0
    for i in range(left, right + 1):
        count = 0
        for j in range(1, int(i ** 0.5) + 1):
            if i % j == 0:
                count += 2
                
            if i ** 0.5 == i // j: # 제곱근이 약수에 포함된 경우 -1 ex) 1, 2, 4
                count -= 1
                
        if count % 2 == 0:
            answer += i
        else:
            answer -= i
    return answer

File: programmers/level1/약수의_개수와_덧셈.cpp
#include <cmath>

using namespace std;

// 단 하나의 수만 비교하도록 만들어 효율성 최대화
// -> 제곱근을 살펴본다

int solution(int left, int right) {
    int answer{ 0 };

    for (int n=left; n<=right; ++n) {
        if (sqrt(n) - (int) sqrt(n) == 0) { // 소수부가 남는가
            answer -= n;
        } else {
            answer += n;
        }
    }

    return answer;
}